kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,386 @@
+backend writers because you can selectively implement methods
+(draw_rectangle, draw_lines, etc...) and slowly see your figure come
+to life w/o having to have a full blown implementation before getting
+any results.
+
+Copy this to backend_xxx.py and replace all instances of 'template'
+with 'xxx'.  Then implement the class methods and functions below, and
+add 'xxx' to the switchyard in matplotlib/backends/__init__.py and
+'xxx' to the backends list in the validate_backend methon in
+matplotlib/__init__.py and you're off.  You can use your backend with::
+
+  import matplotlib
+  matplotlib.use('xxx')
+  from pylab import *
+  plot([1,2,3])
+  show()
+
+matplotlib also supports external backends, so you can place you can
+use any module in your PYTHONPATH with the syntax::
+
+  import matplotlib
+  matplotlib.use('module://my_backend')
+
+where my_backend.py is your module name.  This syntax is also
+recognized in the rc file and in the -d argument in pylab, e.g.,::
+
+  python simple_plot.py -dmodule://my_backend
+
+If your backend implements support for saving figures (i.e. has a print_xyz()
+method) you can register it as the default handler for a given file type
+
+  from matplotlib.backend_bases import register_backend
+  register_backend('xyz', 'my_backend', 'XYZ File Format')
+  ...
+  plt.savefig("figure.xyz")
+
+The files that are most relevant to backend_writers are
+
+  matplotlib/backends/backend_your_backend.py
+  matplotlib/backend_bases.py
+  matplotlib/backends/__init__.py
+  matplotlib/__init__.py
+  matplotlib/_pylab_helpers.py
+
+Naming Conventions
+
+  * classes Upper or MixedUpperCase
+
+  * varables lower or lowerUpper
+
+  * functions lower or underscore_separated
+
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+
+import matplotlib
+from matplotlib._pylab_helpers import Gcf
+from matplotlib.backend_bases import RendererBase, GraphicsContextBase,\
+     FigureManagerBase, FigureCanvasBase
+from matplotlib.figure import Figure
+from matplotlib.transforms import Bbox
+from backend_kivyagg import FigureCanvasKivyAgg
+from matplotlib.backend_bases import ShowBase
+
+try:
+    import kivy
+    from kivy.app import App
+    from kivy.graphics.texture import Texture
+    from kivy.graphics import Rectangle
+    from kivy.uix.widget import Widget
+    from kivy.uix.floatlayout import FloatLayout
+    from kivy.core.window import Window
+    from kivy.base import EventLoop
+except ImportError:
+    raise ImportError("this backend requires Kivy to be installed.")
+
+try:
+    kivy.require('1.9.0') # I would need to check which release of Kivy would be the best suitable.
+except AttributeError:
+    raise ImportError(
+        "kivy version too old -- it must have require_version")
+
+from PIL import Image
+EventLoop.ensure_window()
+
+_debug = True
+
+app = None
+
+def _create_App(fig_canvas):
+    global app
+
+    class TestApp(App):
```

Comment:
This class should be defined outside the function, and instead should take a `fig_canvas` parameter. So you can just do `TestApp(figure=fig_canvas)`.